ang/crmUi.js

```diff
@@ -642,7 +642,7 @@
             $timeout(function () {
               // ex: msg_template_id adds new item then selects it; use $timeout to ensure that
               // new item is added before selection is made
-              element.select2('val', ngModel.$viewValue);
+              element.select2('val', ngModel.$modelValue);
```

Comment:
Toby initially raised issue in [CRM-17192](https://issues.civicrm.org/jira/browse/CRM-17192) which has pretty detailed comments. [Stackoverflow](http://stackoverflow.com/questions/19383812/whats-the-difference-between-ngmodel-modelvalue-and-ngmodel-viewvalue) helped me understand difference. Wish I'd known about it earlier.